neural_speed/models/llama/llama.cpp

```diff
@@ -79,12 +79,6 @@ static bool llama_model_eval_internal(model_context* ctx, const model_input* inp
     n_pasts[i] = inputs[i].n_past;
     n_totals[i] = inputs[i].n_total;
     block_ids[i] = inputs[i].request_idx * beam_size + inputs[i].beam_idx;
-    // enforce that the first token is BOS
-    if (n_totals[i] == 0 && inputs[i].tokens[0] != lctx.vocab.bos_token_id) {
-      fprintf(stderr, "%s: first token must be BOS (token id is %d) in %dth prompt\n", __func__,
-              lctx.vocab.bos_token_id, i);
```

Comment:
why delete here?

Reply:
the tokenizer of llama3 will not set the BOS at the begining of prompts.

Reply:
please keep this, llama3 need bos token, see https://huggingface.co/meta-llama/Meta-Llama-3-8B/discussions/9 and https://github.com/ggerganov/llama.cpp/pull/6751. We can add this instruction in README and updating by following llama3 model hub